src/lib/main/dialog_view.js

```diff
@@ -150,6 +169,158 @@ function updateAutoscanDescription(sliderValue) {
     AutoscanSliderDescriptions[sliderValue];
 }
 
+const ScanModeMap = new Map([
+  ['anywhere', {
+    description: '',
+    options: {
+      partialScan: false,
+      contentMode: Page.contentModeEnum.TEXT,
+      matchMode: Page.matchModeEnum.FIRST,
+    },
+  }],
+  ['inside-elements', {
+    description: `Check only inside selected elements using HTML elements 
+    selector.`,
+    options: {
+      partialScan: true,
+      requireExactMatchCount: true,
+      contentMode: Page.contentModeEnum.TEXT,
+      matchMode: Page.matchModeEnum.FIRST,
+    },
+  }],
+  ['count-only', {
+    description: `Check only for change in number of HTML element matches.
+    Content is ignored.`,
+    options: {
+      partialScan: true,
+      requireExactMatchCount: true,
+      contentMode: Page.contentModeEnum.IGNORE,
+      matchMode: Page.matchModeEnum.FIRST,
+    },
+  }],
+]);
+
+/**
+ * Returns description for selectors input field.
+ *
+ * @param {boolean} partialScan - True if partial scan is active.
+ * @returns {string} Description for selectors.
+ */
+function getSelectorsDescription(partialScan) {
+  if (partialScan) {
+    return `CSS selectors for more information see 
+<a href="https://www.w3schools.com/cssref/css_selectors.asp">
+https://www.w3schools.com/cssref/css_selectors.asp</a>`;
+  } else {
+    return `Selectors not available in "Anywhere" scan mode.`;
```

Comment:
Yes, but it might not be clear why.

Reply:
I think it is pretty clear if the Scan Mode input is shifted to just above the CSS selector field (see my mockup above)

Reply:
Well grouping it under scan mode is really weird and does not make much sense imo since they are totally different things, it just so happens that some scan modes enable/support them. And it's really not clear, I myself wouldn't be sure why it's greyed out, is it cause of a bug? How do I enable it?

I decided to make a compromise so that there is less code around it which seemed like your main goal. It will no longer change the description and instead just set the placeholder, so that new users still immediately know why the feature is disabled and can use the description to look up what it is.